### PR TITLE
Ruby 3 compat drop 2.6

### PR DESCRIPTION
**Pull request by jstoks**
Continuing from the original
PR krisleech/wisper#192 to add support for Ruby 3.0,
this PR simply takes work done by @doits and direction from [the most recent comment](https://github.com/krisleech/wisper/pull/192#issuecomment-1020005293) and attempts to make the requested changes.

Specifically this branch excludes support for ruby 2.6:
[master...doits:b86c57d6064495ef67391405e517dbe85b622d8c](https://github.com/krisleech/wisper/compare/master...doits:b86c57d6064495ef67391405e517dbe85b622d8c)

I also reverted the changes around travis / github actions config.

---

From original PR for context: krisleech/wisper#192
Fixes: krisleech/wisper#190

Makes use of **kwargs everywhere so it is compatible with ruby 3.

Unfortunately this does not work with ruby 2.6 and below, so I dropped it. It could be made to work with 2.6 by using [ruby2_keywords](https://www.ruby-lang.org/en/news/2019/12/12/separation-of-positional-and-keyword-arguments-in-ruby-3-0/#a-compatible-delegation), but I assumend support for 2.6 is not a priority for this gem.

Another thing is I don't know how this works with jruby, e.g. if ruby mri 3 and the use of **kword is compatible to jruby or not.

I hope I didn't miss a place to convert to kword args.

### CODE REVIEW

**Comment by shieldo** [comment]
@krisleech Does this work (on top of https://github.com/krisleech/wisper/pull/192) seem OK to you for a new release? Would be happy to help get things over the line for a new major release.

File: spec/lib/wisper/broadcasters/logger_broadcaster_spec.rb [hunk 1]
```diff
@@ -27,38 +27,41 @@ module Broadcasters
...
           end
 
           context 'with arguments' do
             let(:args) { [arg_double(id: 3), arg_double(id: 4)] }
+            let(:kwargs) { {x: :y} }
 
             it 'logs published event and arguments' do
-              expect(logger).to receive(:info).with('[WISPER] Publisher#1 published thing_created to Listener#2 with Argument#3, Argument#4')
-              subject.broadcast(listener, publisher, event, args)
+              expect(logger).to receive(:info).with("[WISPER] Publisher#1 published thing_created to Listener#2 with Argument#3, Argument#4 and keyword arguments #{kwargs.inspect}")
```
Comment (oskarpearson) [inline spec/lib/wisper/broadcasters/logger_broadcaster_spec.rb:48]: Probably best for your expectation to be hard-coded. That'll match the `Publisher#1` bits in the string too. So something like this (exact output not checked).

```suggestion
              expect(logger).to receive(:info).with("[WISPER] Publisher#1 published thing_created to Listener#2 with Argument#3, Argument#4 and keyword arguments {x: :y}")
```

File: spec/lib/wisper/broadcasters/logger_broadcaster_spec.rb [hunk 2]
```diff
@@ -27,38 +27,41 @@ module Broadcasters
...
 
             context 'when argument is a hash' do
               let(:args) { [hash] }
               let(:hash) { {key: 'value'} }
+              let(:kwargs) { {x: :y} }
 
               it 'logs published event and arguments' do
-                expect(logger).to receive(:info).with("[WISPER] Publisher#1 published thing_created to Listener#2 with Hash##{hash.object_id}: #{hash.inspect}")
-                subject.broadcast(listener, publisher, event, args)
+                expect(logger).to receive(:info).with("[WISPER] Publisher#1 published thing_created to Listener#2 with Hash##{hash.object_id}: #{hash.inspect} and keyword arguments #{kwargs.inspect}")
```
Comment (oskarpearson) [inline spec/lib/wisper/broadcasters/logger_broadcaster_spec.rb:63]: Similar to https://github.com/krisleech/wisper/pull/194/files#r992195618

File: spec/lib/wisper/broadcasters/send_broadcaster_spec.rb [hunk 3]
```diff
@@ -10,7 +10,7 @@ module Broadcasters
 
           it 'sends event to listener without any arguments' do
             expect(listener).to receive(event).with(no_args())
-            subject.broadcast(listener, anything, event, args)
+            subject.broadcast(listener, anything, event, *args)
```
Comment (oskarpearson) [inline spec/lib/wisper/broadcasters/send_broadcaster_spec.rb:13]: Given the title of the spec should this be this instead, where we don't include the args parameter at all? 

```suggestion
            subject.broadcast(listener, anything, event) # leaves out args entirely
```

That may be best done in a different test though?

**Comment by oskarpearson** [comment]
As we're hoping to update our Rails app to Ruby 3 soon *and* add wisper it'd be great if this could get merged 🍾

**Comment by jstoks** [comment]
@oskarpearson I appreciate the feedback and fwiw I think you're right. 

However, I'm no longer working on the project that uses this gem or ruby at all for that matter. I'm going to close this out, but by all means feel free to take it over yourself.

**Pull request closed without merging**